custom/icds_reports/utils/aggregation_helpers/ccs_record_monthly.py

```diff
@@ -52,62 +52,139 @@ def tablename(self):
+                    "({}-case_list.add)::integer<=21)".format(end_month_string, start_month_string)
+        tetanus_complete = "({} AND ut.tt_complete_date is not null AND " \
+                           "ut.tt_complete_date<={})".format(pregnant_to_consider,
+                                                             end_month_string)
+        pnc_complete = "(case_list.pnc1_date<{})".format(end_month_string)
+        bp_visited_in_month = "date_trunc('MONTH', agg_bp.latest_time_end_processed)=" \
+                              "{}".format(start_month_string)
 
         columns = (
-            ('awc_id', 'ucr.awc_id'),
-            ('case_id', 'ucr.case_id'),
-            ('month', 'ucr.month'),
-            ('age_in_months', 'ucr.age_in_months'),
-            ('ccs_status', 'ucr.ccs_status'),
-            ('open_in_month', 'ucr.open_in_month'),
-            ('alive_in_month', 'ucr.alive_in_month'),
-            ('trimester', 'ucr.trimester'),
+            ('awc_id', 'case_list.awc_id'),
+            ('case_id', 'case_list.case_id'),
+            ('month', self.month.strftime("'%Y-%m-%d'")),
+            ('age_in_months', 'trunc({})'.format(age_in_months_end)),
+            ('ccs_status', "CASE WHEN {} THEN 'pregnant' ELSE CASE WHEN {} THEN "
+                           "'lactating' ELSE 'other' END END".format(pregnant_to_consider,
+                                                                     lactating)),
+            ('open_in_month', "CASE WHEN {} THEN 1 ELSE 0 END".format(open_in_month)),
+            ('alive_in_month', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(alive_in_month)),
+            ('trimester', trimester),
             ('num_rations_distributed', 'COALESCE(agg_thr.days_ration_given_mother, 0)'),
-            ('thr_eligible', 'ucr.thr_eligible'),
-            ('tetanus_complete', 'ucr.tetanus_complete'),
-            ('delivered_in_month', 'ucr.delivered_in_month'),
-            ('anc1_received_at_delivery', 'ucr.anc1_received_at_delivery'),
-            ('anc2_received_at_delivery', 'ucr.anc2_received_at_delivery'),
-            ('anc3_received_at_delivery', 'ucr.anc3_received_at_delivery'),
-            ('anc4_received_at_delivery', 'ucr.anc4_received_at_delivery'),
-            ('registration_trimester_at_delivery', 'ucr.registration_trimester_at_delivery'),
-            ('using_ifa', 'ucr.using_ifa'),
-            ('ifa_consumed_last_seven_days', 'ucr.ifa_consumed_last_seven_days'),
-            ('anemic_severe', 'ucr.anemic_severe'),
-            ('anemic_moderate', 'ucr.anemic_moderate'),
-            ('anemic_normal', 'ucr.anemic_normal'),
-            ('anemic_unknown', 'ucr.anemic_unknown'),
-            ('extra_meal', 'ucr.extra_meal'),
-            ('resting_during_pregnancy', 'ucr.resting_during_pregnancy'),
-            ('bp_visited_in_month', 'ucr.bp_visited_in_month'),
+            ('thr_eligible', 'CASE WHEN {} THEN 1 ELSE 0 END'.format(valid_in_month)),
```

Comment:
you only have to be valid in month to be thr eligible? why was it a separate ucr field before?

Reply:
so valid in month is made up of is (pregnant or lactating) which further are broken down to more  parameters . here: https://github.com/dimagi/commcare-hq/pull/22752/files/6de226f5336b42ca7d4a5a32741dcd07179f1cf4#diff-3fdee4e410a43f289000bcdf4802c62aR82.

The reason for It to be  a separate UCR field before is not clear to me. It seems redundant to me to store it as UCR field. One Reason I could think of is that earlier no calculation was done outside of UCR.